File: 프로그래머스/lv3/이현섭/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    vector<int> answer;
    int now = s/n;
    if(now<1)
        return {-1};
    
    for(int i=0;i<n;i++)
    {
        answer.push_back(now);
    }
    for(int i=0;i<s%n;i++)
    {
        answer[n-i-1]+=1;
    }
    
    return answer;
}

File: 프로그래머스/lv3/이재표/최고의 집합.java
import java.util.*;
class Solution {
    public int[] solution(int n, int s) {
        int[] answer = new int[n];
        if(s/n==0){
            return new int[]{-1};
        }else{
            int a=s/n;
            int b=s%n;
            for(int i=0;i<n;i++){
                answer[i]=a; 
            }
            for(int i=0;i<b;i++){
                answer[n-i-1]++;
            }
        }
        return answer;
    }
}
